### PR TITLE
GCE provider

### LINKED ISSUE #406: Missing documentation for GCE
A page should be added to 1.23 (and later) for the GCE provider.  Let me know if you need more info than what's in the 1.23 release notes.

### PR DESCRIPTION
**Pull request by evilnick**
New docs for GCE provider
fixes #406

### CODE REVIEW

**Comment by evilnick** [comment]
@petermatulis Some notes for reviewing:

* you can check out the changes and make comments inline above
* Do check both for wording/explanations and technical details 
* If it weren't me submitting this, I would also suggest verifying any steps/commands actually work (they do in this case!)
* You can fetch this entire branch easily from this link https://github.com/evilnick/docs/archive/gce-provider.zip (or otherwise, visiting my fork of the docs) and build it locally to check rendering
* To build local docs, instructions are in the readme, but basically:
``` make build
    make serve
```
... will render an almost as good as real version of the docs

File: src/en/config-gce.md [hunk 1]
```diff
@@ -0,0 +1,227 @@
+Title: Configuring Juju for Google Compute Engine
+
+# Configuring Juju for Google Compute Engine
+
```
Comment (petermatulis) [inline src/en/config-gce.md]: THIS IS A TEST

File: src/en/config-gce.md [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+**Note:** The above command will not overwrite your existing environments.yaml
+file, or output to stdout. In order to see the boilerplate environments.yaml on
+stdout you need to append the `--show` option. This is helpful if you have an
+existing environments.yaml and just need to add a section. E.g. 
+`juju generate-config --show`. You can then copy and paste the needed section.
+
+The generic configuration section generated for Windows Azure will look
+something like this:
+
```
Comment (petermatulis) [inline src/en/config-gce.md]: s/Windows Azure/Google Compute Engine

File: src/en/config-gce.md [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+      # provision. While the region has a default, the project ID is
+      # required. For information on the project ID, see
+      # https://cloud.google.com/compute/docs/projects and regarding regions
+      # see https://cloud.google.com/compute/docs/zones.
+      project-id:
+      # region: us-central1
+    
+      # The GCE provider uses pre-built images when provisioning instances.
+      # You can customize the location in which to find them with the
+      # image-endpoint setting. The default value is the a location within
```
Comment (petermatulis) [inline src/en/config-gce.md]: "The default value is a location within..."

File: src/en/config-gce.md [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+      # GCE, so it will give you the best speed when bootstrapping or adding
+      # machines. For more information on the image cache see
+      # https://cloud-images.ubuntu.com/.
+      # image-endpoint: https://www.googleapis.com
+```
+
+In order to find out the necessary values to add to the configuration, you will
```
Comment (petermatulis) [inline src/en/config-gce.md]: "To determine configuration values you will..."

File: src/en/config-gce.md (hunk 3 above)
Comment (evilnick) [inline src/en/config-gce.md]: yeah - this is from the boilerplate generated by juju, so we sort of have to use it warts and all until I submit a fix to juju core

File: src/en/config-gce.md (hunk 4 above)
Comment (evilnick) [inline src/en/config-gce.md]: hmmm. I like it that it explicitly says you are adding things to the configuration. How about 
"To determine the values to add..."?

**Comment by petermatulis** [comment]
:+1:

**Pull request merged**